File: app/src/main/java/knf/kuma/preferences/BottomPreferencesFragment.java
package knf.kuma.preferences;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import knf.kuma.BottomFragment;
import knf.kuma.R;
import knf.kuma.commons.EAHelper;

public class BottomPreferencesFragment extends BottomFragment {
    public BottomPreferencesFragment() {
    }

    public static BottomPreferencesFragment get(){
        return new BottomPreferencesFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        EAHelper.enter1("C");
        return inflater.inflate(R.layout.fragment_preferences,container,false);
    }

    @Override
    public void onReselect() {
        EAHelper.enter1("C");
    }
}

File: app/src/main/java/knf/kuma/preferences/BottomPreferencesFragment.kt
package knf.kuma.preferences

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import knf.kuma.BottomFragment
import knf.kuma.R
import knf.kuma.achievements.AchievementManager
import knf.kuma.commons.EAHelper

class BottomPreferencesFragment : BottomFragment() {

    var count = 1

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        EAHelper.enter1("C")
        return inflater.inflate(R.layout.fragment_preferences, container, false)
    }

    override fun onReselect() {
        EAHelper.enter1("C")
        count++
        if (count == 20) AchievementManager.unlock(listOf(40))
    }

    companion object {

        fun get(): BottomPreferencesFragment {
            return BottomPreferencesFragment()
        }
    }
}
